src/Common/tests/System/IO/FileCleanupTestBase.cs

```diff
@@ -18,15 +18,25 @@ protected FileCleanupTestBase()
             // so that it can be easily correlated if necessary, and the random string to helps avoid conflicts if
             // the same test should be run concurrently with itself (e.g. if a [Fact] method lives on a base class)
             // or if some stray files were left over from a previous run.
-            TestDirectory = Path.Combine(Path.GetTempPath(), GetType().Name + "_" + Path.GetRandomFileName());
-            try
-            {
-                Directory.CreateDirectory(TestDirectory);
-            }
-            catch (Exception ex)
+
+            // Make 3 attempts since we have seen this on rare occasions fail with access denied, perhaps due to machine
+            // configuration, and it doesn't make sense to fail arbitrary tests for this reason.
+            string failure = String.Empty;
+            for (int i = 0; i <= 2; i++)
             {
-                Assert.True(false, $"FileCleanupTestBase failed to create {TestDirectory} due to {ex.ToString()}");
+                TestDirectory = Path.Combine(Path.GetTempPath(), GetType().Name + "_" + Path.GetRandomFileName());
+                try
+                {
+                    Directory.CreateDirectory(TestDirectory);
+                    break;
+                }
+                catch (Exception ex)
+                {
+                    failure += ex.ToString() + "\n";
```

Comment:
You may also sleep 10 milliseconds so whatever intermittent machine condition triggers the problem has a chance to go away.